src/Utils/request/uploadFile.ts

```diff
@@ -71,4 +71,81 @@ const uploadFile = async (
   });
 };
 
+export const uploadMultipleFiles = async (
+  files: File[],
+  createUploadFn: (file: File, index: number) => Promise<any>,
+  markUploadCompleteFn: (args: {
+    data: any;
+    associating_id: string;
+  }) => Promise<any>,
+  options: {
+    associating_id: string;
+    setProgress: (progress: number) => void;
+    setError: (msg: string) => void;
+  },
+): Promise<{ errors: File[] }> => {
+  const { associating_id, setProgress, setError } = options;
+
+  const totalBytes = files.reduce((sum, f) => sum + f.size, 0);
+  let completedFilesBytes = 0;
+  const errors: File[] = [];
+
+  const updateProgress = (currentFileBytes: number) => {
+    const totalUploadedBytes = completedFilesBytes + currentFileBytes;
+    setProgress(totalBytes > 0 ? (totalUploadedBytes / totalBytes) * 100 : 0);
+  };
+
+  for (const [index, file] of files.entries()) {
+    try {
+      const data = await createUploadFn(file, index);
+      if (data) {
+        await new Promise<void>((resolve, reject) => {
+          uploadFile(
```

Comment:
I know this is what we have right now, but go ahead and clean this up a little bit. Create separate functions like getSuccessHandler/getErrorHandler etc that encapsulates success/error logic. This should make it readable.

Same for uploadMultipleFiles call in useFileUpload.